fml/thread.cc

```diff
@@ -0,0 +1,66 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/fml/thread.h"
+#include <memory>
+#include <string>
+#include "flutter/fml/message_loop.h"
+#include "lib/ftl/build_config.h"
+#include "lib/ftl/synchronization/waitable_event.h"
+
+#if OS_MACOSX
+#include <pthread/pthread.h>
+#elif OS_LINUX
+#include <pthread.h>
+#else
+#error Unsupported Platform
+#endif
+
+namespace fml {
+
+Thread::Thread(const std::string& name) : joined_(false) {
+  ftl::AutoResetWaitableEvent latch;
+  ftl::RefPtr<ftl::TaskRunner> runner;
+  thread_ = std::make_unique<std::thread>([&latch, &runner, name]() -> void {
+    SetCurrentThreadName(name);
+    auto& loop = MessageLoop::GetCurrent();
+    runner = loop.GetTaskRunner();
+    latch.Signal();
+    loop.Run();
+  });
+  latch.Wait();
```

Comment:
Currently, the task runner holds a reference to the message loop it posts tasks to. There is currently no provision to create a task runner and later associate it with a message loop. That is not to say that cannot be done. I will attempt that in a later patch if that is OK.